apps/dashr-clinical-analytics/app.R

```diff
@@ -0,0 +1,492 @@
+library(dashR)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(plotly)
+library(chron)
+library(dplyr)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != ""){
+  pathPrefix <- sprintf("/%s/", appName)
+
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+  setwd(sprintf("./app/apps/%s", appName))
+}
+
+app <- Dash$new()
+
+setwd(sprintf("./app/apps/%s", appName))
+
+df <- read.csv('./data/clinical_analytics.csv', header = TRUE, sep = ",")
+
+
+#list of clinics
+clinics = unique(df$Clinic.Name)
+clinic_list <- list()
+for (i in 1:length(clinics)){
+  clinic_list[[i]] <- list(label = clinics[i], value = clinics[i])
+}
+
+#list of check in times
+times = unique(df$Check.In.Time)
+times_list <- list()
```

Comment:
I filled the lists using `as.list(unique(df$...))` :)

Reply:
Right. My comment was more about filling in objects of known dimension up front, like `data.frame` and sometimes lists of `data.frame` objects rather than those of arbitrary dimension. R is not incredibly memory efficient, so this can help.

`lapply` and friends are basically abstracting away the loop, but in an intelligent way. You can always create the empty list up front, but the nice thing about `lapply` is that it’s returned for free.